### PR TITLE
Select component updates

### PR DESCRIPTION
**Pull request by pabloai**
## Summary
- Make it possible to filter options in the Select component
- Update component to use hooks.
- Filterable lists will be the new default behaviour.
- Added strings:
```
"select.noOptionsFound": "No options found"
"select.filter.placeholder": "Search..."
```

## Tested scenarios
Both filterable and non filterable lists.

**Fixed issue**:  Adyen/adyen-web#725

### CODE REVIEW

File: packages/lib/src/language/locales/en-US.json [hunk 1]
```diff
@@ -135,6 +135,8 @@
     "select.stateOrProvince": "Select state or province",
     "select.provinceOrTerritory": "Select province or territory",
     "select.country": "Select country",
+    "select.noOptionsFound": "No options found",
+    "select.filter.placeholder": "Search...",
```
Comment (marcperez) [inline packages/lib/src/language/locales/en-US.json:139]: Reminder: these need to be added into the translations repository

File: packages/lib/src/components/internal/FormFields/Select/Select.tsx [hunk 2]
```diff
@@ -1,217 +1,201 @@
...
+
+    const active: SelectItem = props.items.find(i => i.id === props.selected) || ({} as SelectItem);
+
+    const keys = {
+        arrowDown: 'ArrowDown',
+        arrowUp: 'ArrowUp',
+        enter: 'Enter',
+        escape: 'Escape',
+        space: ' '
     };
```
Comment (marcperez) [inline packages/lib/src/components/internal/FormFields/Select/Select.tsx:29]: This could also be declared outside of this component, and imported here.

**Review by marcperez — APPROVED** [review]

File: packages/lib/src/components/internal/FormFields/Select/Select.tsx (hunk 2 above)
Comment (pabloai) [reply packages/lib/src/components/internal/FormFields/Select/Select.tsx]: Done.

File: packages/lib/src/language/locales/en-US.json (hunk 1 above)
Comment (pabloai) [reply packages/lib/src/language/locales/en-US.json:139]: I've created the translation job.

**Review by marcperez — APPROVED** [review]

**Pull request merged**